core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4885,16 +4903,36 @@ public static boolean isCheckURIEncodingEnabled() {
         return ExtensionList.lookupSingleton(URICheckEncodingMonitor.class).isCheckEnabled();
     }
 
+
+    public Future<DependencyGraph> getFutureDependencyGraph() {
+        synchronized (dependencyGraphLock) {
+            //Scheduled future will be the most recent one --> Return
+            if (scheduledFutureDependencyGraph != null) {
+                return scheduledFutureDependencyGraph;
+            }
+            //Calculating future will be the most recent one --> Return
+            if (calculatingFutureDependencyGraph != null) {
+                return calculatingFutureDependencyGraph;
+            }
+            //No scheduled or calculating future --> Already completed dependency graph is the most recent one
+            return CompletableFuture.completedFuture(dependencyGraph);
+        }
+    }
+
+
+    public DependencyGraph createNewDependencyGraph() {
```

Comment:
Probably just inline this call into its callers, we don't want to add this new public API since no one would find it useful. If you need it for testing could you make it package scope?

Reply:
I couldn't make it package scope. The relevant unit-test is located in jenkins-test, because I need JenkinsRule which is only available in jenkins-test.  

By modifying the unit-test, I could do without the method. 

Thanks for your feedback.